pkg/reconciler/certificate/certificate_test.go

```diff
@@ -138,6 +138,39 @@ func TestReconcile(t *testing.T) {
 			Eventf(corev1.EventTypeNormal, "Updated", "Updated Spec for Cert-Manager Certificate %s/%s", "foo", "knCert"),
 		},
 		Key: "foo/knCert",
+	}, {
+		Name: "observed generation is still updated when error is encountered",
+		Objects: []runtime.Object{
+			knCertWithGeneration("knCert", "foo", generation+1),
```

Comment:
What happens if the status of the initial knCert has condition Ready=True instead of nil? That's the case I was worried about in [this comment](https://github.com/knative/serving/pull/5207#issuecomment-522736271).

Reply:
@jonjohnsonjr I see. Would marking the knCert as not ready solve this problem?